src/renderer/component/fileDownloadLink/view.jsx

```diff
@@ -1,19 +1,39 @@
+// @flow
 import React from 'react';
 import Button from 'component/button';
-import classnames from 'classnames';
 import * as icons from 'constants/icons';
 
-class FileDownloadLink extends React.PureComponent {
+type Props = {
+  uri: string,
+  downloading: boolean,
+  fileInfo: ?{
+    written_bytes: number,
+    total_bytes: number,
+    outpoint: number,
+    download_path: string,
+  },
+  loading: boolean,
+  costInfo: ?{},
+  restartDownload: (string, number) => void,
+  checkAvailability: string => void,
+  openInShell: string => void,
+  purchaseUri: string => void,
+  doPause: () => void,
+};
+
+class FileDownloadLink extends React.PureComponent<Props> {
   componentWillMount() {
     this.checkAvailability(this.props.uri);
   }
 
-  componentWillReceiveProps(nextProps) {
+  componentWillReceiveProps(nextProps: Props) {
```

Comment:
componentWillReceiveProps is being deprecated in favor of getDerivedStateFromProps